File: DataBindingLiveDataJava/app/src/main/java/com/websarva/wings/android/databindinglivedatajava/MainActivity.java
package com.websarva.wings.android.databindinglivedatajava;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import com.websarva.wings.android.databindinglivedatajava.databinding.ActivityMainBinding;
import com.websarva.wings.android.databindinglivedatajava.ui.MainViewModel;

public class MainActivity extends AppCompatActivity {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		ActivityMainBinding activityMainBinding = DataBindingUtil.setContentView(MainActivity.this, R.layout.activity_main);
		ViewModelProvider provider = new ViewModelProvider(MainActivity.this);
		MainViewModel mainViewModel = provider.get(MainViewModel.class);
		activityMainBinding.setLifecycleOwner(MainActivity.this);
		activityMainBinding.setMainViewModel(mainViewModel);
	}
}

File: DataBindingLiveDataKotlin/app/src/main/java/com/websarva/wings/android/databindinglivedatakotlin/MainActivity.kt
package com.websarva.wings.android.databindinglivedatakotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.activity.viewModels
import androidx.databinding.DataBindingUtil
import com.websarva.wings.android.databindinglivedatakotlin.databinding.ActivityMainBinding
import com.websarva.wings.android.databindinglivedatakotlin.ui.MainViewModel

class MainActivity : AppCompatActivity() {
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		val activityMainBinding = DataBindingUtil.setContentView<ActivityMainBinding>(this@MainActivity, R.layout.activity_main)
		val mainViewModel by viewModels<MainViewModel>()
		activityMainBinding.mainViewModel = mainViewModel
		activityMainBinding.lifecycleOwner = this@MainActivity
	}
}
